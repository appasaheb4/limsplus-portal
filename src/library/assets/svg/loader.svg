<svg id="eg7Z31GJonW1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 75 50" shape-rendering="geometricPrecision" text-rendering="geometricPrecision">
<style><![CDATA[
#eg7Z31GJonW5_to {animation: eg7Z31GJonW5_to__to 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW5_to__to { 0% {transform: translate(70px,65px)} 25% {transform: translate(70px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 50% {transform: translate(100px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 75% {transform: translate(130px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 100% {transform: translate(130px,65px)}} #eg7Z31GJonW5_ts {animation: eg7Z31GJonW5_ts__ts 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW5_ts__ts { 0% {transform: scale(0,0);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 25% {transform: scale(5,5);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 75% {transform: scale(5,5);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 100% {transform: scale(0,0)}} #eg7Z31GJonW6_to {animation: eg7Z31GJonW6_to__to 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW6_to__to { 0% {transform: translate(130px,65px)} 20% {transform: translate(130px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 30% {transform: translate(70px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 50% {transform: translate(70px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 75% {transform: translate(100px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 100% {transform: translate(130px,65px)}} #eg7Z31GJonW6_ts {animation: eg7Z31GJonW6_ts__ts 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW6_ts__ts { 0% {transform: scale(5,5);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 20% {transform: scale(0,0);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 31.5% {transform: scale(0,0);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 50% {transform: scale(5,5)} 100% {transform: scale(5,5)}} #eg7Z31GJonW7_to {animation: eg7Z31GJonW7_to__to 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW7_to__to { 0% {transform: translate(100px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 25% {transform: translate(130px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 45% {transform: translate(130px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 55% {transform: translate(70px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 75% {transform: translate(70px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 100% {transform: translate(100px,65px)}} #eg7Z31GJonW7_ts {animation: eg7Z31GJonW7_ts__ts 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW7_ts__ts { 0% {transform: scale(5,5)} 25% {transform: scale(5,5);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 45% {transform: scale(0,0);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 56.5% {transform: scale(0,0);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 75% {transform: scale(5,5);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 100% {transform: scale(5,5)}} #eg7Z31GJonW8_to {animation: eg7Z31GJonW8_to__to 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW8_to__to { 0% {transform: translate(70px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 25% {transform: translate(100px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 50% {transform: translate(130px,65px);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 70% {transform: translate(130px,65px);animation-timing-function: cubic-bezier(0.42,0,0.55,0.99)} 75% {transform: translate(70px,65px)} 100% {transform: translate(70px,65px)}} #eg7Z31GJonW8_ts {animation: eg7Z31GJonW8_ts__ts 2000ms linear infinite normal forwards}@keyframes eg7Z31GJonW8_ts__ts { 0% {transform: scale(5,5)} 50% {transform: scale(5,5);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 66% {transform: scale(0,0);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 82.5% {transform: scale(0,0);animation-timing-function: cubic-bezier(0.4,0.005,0.515,0.98)} 100% {transform: scale(5,5)}}
]]></style>
<g transform="translate(-62.499996-40)"><g id="eg7Z31GJonW5_to" transform="translate(70,65)"><g id="eg7Z31GJonW5_ts" transform="scale(0,0)"><circle r="1.38" transform="translate(0,0)" fill="#C7222A"/></g></g><g id="eg7Z31GJonW6_to" transform="translate(130,65)"><g id="eg7Z31GJonW6_ts" transform="scale(5,5)"><circle r="1.38" transform="translate(0,0)" fill="#C7222A"/></g></g><g id="eg7Z31GJonW7_to" transform="translate(100,65)"><g id="eg7Z31GJonW7_ts" transform="scale(5,5)"><circle r="1.38" transform="translate(0,0)" fill="#C7222A"/></g></g><g id="eg7Z31GJonW8_to" transform="translate(70,65)"><g id="eg7Z31GJonW8_ts" transform="scale(5,5)"><circle r="1.38" transform="translate(0,0)" fill="#C7222A"/></g></g></g></svg>
