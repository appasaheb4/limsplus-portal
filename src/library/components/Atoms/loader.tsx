import React from "react";
import * as Bootstrap from "react-bootstrap";
const Loader = () => {
  return (
    <div className="bg-black">
      <Bootstrap.Modal show={true} centered>
        <Bootstrap.Modal.Body style={{ textAlign: "center" }}>
          <Bootstrap.Spinner animation="border" variant="dark" />
          <br />
          <span>Loading...</span>
        </Bootstrap.Modal.Body>
      </Bootstrap.Modal>
    </div>
  );
};

export default Loader;
