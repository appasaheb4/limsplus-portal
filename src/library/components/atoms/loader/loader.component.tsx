import React from 'react';
import * as Bootstrap from 'react-bootstrap';
export const Loader = () => {
  return (
    <div className="bg-black">
      <Bootstrap.Modal show={true} centered>
        <Bootstrap.Modal.Body style={{textAlign: 'center'}}>
          <Bootstrap.Spinner animation="border" variant="dark" />
          <br />
          <span>Loading...</span>
        </Bootstrap.Modal.Body>
      </Bootstrap.Modal>
    </div>
  );
};

export const ModalLoader = () => {
  return (
    <>
      <>
        <div className="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none">
          <div className="relative w-full my-6 mx-auto max-w-xl">
            {/*content*/}
            <div className="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none">
              {/*body*/}
              <div className="relative p-2 flex-auto flex flex-col items-center">
                <style>
                  {`
    .spinner .background {
      fill: #555;
    }
    .spinner .line {
      animation: PacMan 5s infinite;
      fill: none;
      stroke: #d26188;
      stroke-width: 25;
    }
    .spinner .spinner {
      animation: Spin 2s infinite linear;
    }
    @keyframes PacMan {
      0% {
        stroke-dasharray: 79px 79;
      }
      50% {
        stroke-dasharray: 1px 79;
      }
      100% {
        stroke-dasharray: 79px 79;
      }
    }
    @keyframes Spin {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
    `}
                </style>
                <svg
                  className="spinner"
                  width="100"
                  height="100"
                  viewBox="0 0 100 100"
                >
                  <circle className="background" cx="0" cy="0"></circle>
                  <path
                    className="line"
                    d="M 37.5,50 C 37.5,43.096441 43.096441,37.5 50,37.5 C 56.903559,37.5 62.5,43.096441 62.5,50 C 62.5,56.903559 56.903559,62.5 50,62.5 C 43.096441,62.5 37.5,56.903559 37.5,50"
                  ></path>
                </svg>{' '}
                <span style={{marginTop: -15}}>loading ...</span>
              </div>
            </div>
          </div>
        </div>
        <div className="opacity-25 fixed inset-0 z-40 bg-black"></div>
      </>
    </>
  );
};
