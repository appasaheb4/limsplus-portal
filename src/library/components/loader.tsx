import React from 'react';
import * as Bootstrap from 'react-bootstrap';
const Loader = () => {
  return (
    <Bootstrap.Modal show={true} centered>
      <Bootstrap.Modal.Body style={{ textAlign: 'center' }}>
        <Bootstrap.Spinner animation="border" variant="dark" />
        <br />
        <span>Loading...</span>
      </Bootstrap.Modal.Body>
    </Bootstrap.Modal>
  );
};

export default Loader;
