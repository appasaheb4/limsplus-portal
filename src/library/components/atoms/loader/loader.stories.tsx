import React from 'react';
import {ComponentStory, ComponentMeta} from '@storybook/react';
import {Loader, ModalLoader} from './loader.component';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Atoms/Loader',
  component: Loader,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: {control: 'color'},
  },
} as ComponentMeta<typeof Loader>;

export const _Loader: ComponentStory<typeof Loader> = () => <Loader />;

export const _ModalLoader: ComponentStory<typeof Loader> = () => (
  <ModalLoader />
);
