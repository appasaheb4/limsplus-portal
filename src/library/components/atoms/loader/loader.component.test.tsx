import React from 'react';
import {render} from '@testing-library/react';
import {Loader, ModalLoader} from './loader.component';

describe('Loader component', () => {
  it('render loader correctly', () => {
    const loader = render(<Loader />);
    expect(loader).toMatchSnapshot();
  });
});

describe('Modal Loader component', () => {
  it('render modal loader correctly', () => {
    const modalLoader = render(<ModalLoader />);
    expect(modalLoader).toMatchSnapshot();
  });
});
