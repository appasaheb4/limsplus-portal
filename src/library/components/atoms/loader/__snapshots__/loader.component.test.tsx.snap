// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component render loader correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="bg-black"
      />
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-body"
            style="text-align: center;"
          >
            <div
              class="spinner-border text-dark"
            />
            <br />
            <span>
              Loading...
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="bg-black"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Loader component render modal loader correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class=""
    style="padding-right: 0px;"
  >
    <div>
      <div
        class="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
      >
        <div
          class="relative  mx-auto "
        >
          <div
            class="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none"
          >
            <div
              class="relative p-2 flex-auto flex flex-col items-center"
            >
              <style>
                
    .spinner .background {
      fill: #555;
    }
    .spinner .line {
      animation: PacMan 5s infinite;
      fill: none;
      stroke: #d26188;
      stroke-width: 25;
    }
    .spinner .spinner {
      animation: Spin 2s infinite linear;
    }
    @keyframes PacMan {
      0% {
        stroke-dasharray: 79px 79;
      }
      50% {
        stroke-dasharray: 1px 79;
      }
      100% {
        stroke-dasharray: 79px 79;
      }
    }
    @keyframes Spin {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
    
              </style>
              <svg
                class="spinner"
                height="100"
                viewBox="0 0 100 100"
                width="100"
              >
                <circle
                  class="background"
                  cx="0"
                  cy="0"
                />
                <path
                  class="line"
                  d="M 37.5,50 C 37.5,43.096441 43.096441,37.5 50,37.5 C 56.903559,37.5 62.5,43.096441 62.5,50 C 62.5,56.903559 56.903559,62.5 50,62.5 C 43.096441,62.5 37.5,56.903559 37.5,50"
                />
              </svg>
               
              <span
                style="margin-top: -15px;"
              >
                loading ...
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="opacity-25 fixed inset-0 z-40 bg-black"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
    >
      <div
        class="relative  mx-auto "
      >
        <div
          class="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none"
        >
          <div
            class="relative p-2 flex-auto flex flex-col items-center"
          >
            <style>
              
    .spinner .background {
      fill: #555;
    }
    .spinner .line {
      animation: PacMan 5s infinite;
      fill: none;
      stroke: #d26188;
      stroke-width: 25;
    }
    .spinner .spinner {
      animation: Spin 2s infinite linear;
    }
    @keyframes PacMan {
      0% {
        stroke-dasharray: 79px 79;
      }
      50% {
        stroke-dasharray: 1px 79;
      }
      100% {
        stroke-dasharray: 79px 79;
      }
    }
    @keyframes Spin {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
    
            </style>
            <svg
              class="spinner"
              height="100"
              viewBox="0 0 100 100"
              width="100"
            >
              <circle
                class="background"
                cx="0"
                cy="0"
              />
              <path
                class="line"
                d="M 37.5,50 C 37.5,43.096441 43.096441,37.5 50,37.5 C 56.903559,37.5 62.5,43.096441 62.5,50 C 62.5,56.903559 56.903559,62.5 50,62.5 C 43.096441,62.5 37.5,56.903559 37.5,50"
              />
            </svg>
             
            <span
              style="margin-top: -15px;"
            >
              loading ...
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="opacity-25 fixed inset-0 z-40 bg-black"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
